packages/vehicle-rental-overlay/src/leaflet-layer-control-interface.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react";
+
+import VehicleRentalOverlay from ".";
+
+type Props = {
+  /**
+   * This method is created by the @opentripplanner/base-map package when
+   * mounting user-defined layers. This will allow the component to subscribe to
+   * various leaflet events that then get forwarded on to this component.
+   */
+  registerOverlay: (component: Component) => void;
+  /**
+   * Whether or not to initially display the rental vehicles. Once the component
+   * is mounted the subscribed events of a user toggling on or off the layer via
+   * the layer control item will subsequently control the display of the
+   * component.
+   */
+  visible?: boolean;
+};
+
+type State = {
+  /**
+   * An internal state variable used to determine whether or not to display the
+   * rental vehicles for this layer.
+   */
+  visible: boolean;
+};
+
+/**
+ * This class is a necessary intermediary to handle events from the Leaflet map
+ * existing in the parent element. See more notes on the other properties and
+ * methods for how everything interacts.
+ */
+export default class LeafletLayerControlInterface extends Component<
```

Comment:
(non-blocking) There seem to be two opposite approaches, one taken in OTP-UI by e.g. `ParkAndRideOverlay`, `StopsOverlay`, and `TransitVehicleOverlay` that doesn't keep the `visible` state of the layer, and this one, also used in OTP-RR by `ConnectedVehicleRentalOverlay` and `GtfsRtVehicleOverlay`... I am still missing something about the two approaches, but long term I would favor an approach similar to `ParkAndRideOverlay`.